test/e2e/framework/pod/delete.go

```diff
@@ -57,10 +57,17 @@ func DeletePodWithWait(c clientset.Interface, pod *v1.Pod) error {
 }
 
 // DeletePodWithWaitByName deletes the named and namespaced pod and waits for the pod to be terminated. Resilient to the pod
-// not existing.
+// not existing. Also waits for all owned resources to be deleted.
 func DeletePodWithWaitByName(c clientset.Interface, podName, podNamespace string) error {
 	e2elog.Logf("Deleting pod %q in namespace %q", podName, podNamespace)
-	err := c.CoreV1().Pods(podNamespace).Delete(context.TODO(), podName, metav1.DeleteOptions{})
+	deletionPolicy := metav1.DeletePropagationForeground
+	err := c.CoreV1().Pods(podNamespace).Delete(context.TODO(), podName,
+		metav1.DeleteOptions{
+			// If the pod is the owner of some resources (like ephemeral inline volumes),
+			// then we want to be sure that those are also gone before we return.
+			// Blocking pod deletion via metav1.DeletePropagationForeground achieves that.
+			PropagationPolicy: &deletionPolicy,
```

Comment:
I'm not sure switching this to foreground propagation for all existing callers was a good idea... it puts GC in the critical path and requires kube-controller-manager to be running (which I'm not sure it does in all e2e jobs, xref https://github.com/kubernetes/kubernetes/issues/93302#issuecomment-661980020)